src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { MpcWalletContext } from '@/components/common/ConnectWallet/MPCWalletProvider'
+import CopyButton from '@/components/common/CopyButton'
+import ModalDialog from '@/components/common/ModalDialog'
+import { Box, Button, DialogContent, DialogTitle, IconButton, TextField, Typography } from '@mui/material'
+import { useContext, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { Visibility, VisibilityOff, Close } from '@mui/icons-material'
+import css from './styles.module.css'
+import ErrorCodes from '@/services/exceptions/ErrorCodes'
+import { logError } from '@/services/exceptions'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+
+enum ExportFieldNames {
+  password = 'password',
+  pk = 'pk',
+}
+
+type ExportFormData = {
+  [ExportFieldNames.password]: string
+  [ExportFieldNames.pk]: string | undefined
+}
+
+const ExportMPCAccountModal = ({ onClose, open }: { onClose: () => void; open: boolean }) => {
+  const { exportPk } = useContext(MpcWalletContext)
+  const [error, setError] = useState<string>()
+
+  const [showPassword, setShowPassword] = useState(false)
+  const formMethods = useForm<ExportFormData>({
+    mode: 'all',
+    defaultValues: {
+      [ExportFieldNames.password]: '',
+    },
+  })
+  const { register, formState, handleSubmit, setValue, watch, reset } = formMethods
+
+  const exportedKey = watch(ExportFieldNames.pk)
+
+  const onSubmit = async (data: ExportFormData) => {
+    try {
+      setError(undefined)
+      const pk = await exportPk(data[ExportFieldNames.password])
+      setValue(ExportFieldNames.pk, pk)
+    } catch (err) {
+      logError(ErrorCodes._305, err)
+      setError('Error exporting account. Your entered password might be invalid.')
```

Comment:
Is it possible to reuse the same string that we throw from `exportPk`? I would slightly reword it to sound more active e.g. `Error exporting account. Make sure the password is correct.`